### ISSUE TITLE
Stable Diffusion x4 upscaler has a different expected value range than other pipelines

### ISSUE THREAD

**Issue by yiyixuxu** [huggingface/diffusers#3654]
for input/output in `pt` format, `StableDiffusionUpscalePipeline` expect the input to have value range `[-1,1]` and also output `[-1,1]`

see `preprocess` method here, i.e. It will not apply normalization if the input is a pytorch tensor
https://github.com/huggingface/diffusers/blob/94a0c644a8ce5b05a969859e0814ef4883ac870e/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_upscale.py#L37

and `postprocess`, it will not apply denormalization if we output a pytorch tensor
https://github.com/huggingface/diffusers/blob/94a0c644a8ce5b05a969859e0814ef4883ac870e/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_upscale.py#L735

A little bit of context is that for all the pipelines that we've refactored with `VaeImageProcessor`, we now expect all input and outputs to be in `[0,1]` value range. And, previously before the refactoring, all the pipelines expect pytorch tensor input to be in `[-1,1]` but output in `[0,1]` - this inconsistency is one of the reasons we started to create this class to unify the image processing API. 

Since we want to be able to chain the pipelines, we would want to have consistent value ranges among all the pipelines no? that way we can pass the PyTorch tensor output from one pipeline to another seamlessly. 

Also I found this commit that made the x4 upscaler to output pytorch tensors in `[-1,1]` @patrickvonplaten 
https://github.com/huggingface/diffusers/commit/da2ce1a6b92f48cabe9e9d3944c4ee8b007b2871

**Comment by yiyixuxu** [comment]
@patrickvonplaten 

I don't think it's a bug - it is more about whether we want to keep our expected input and output range consistent between pipelines. 

Here is the test 
```python
import requests
from PIL import Image
from io import BytesIO
from diffusers import StableDiffusionUpscalePipeline
import torch
import numpy as np

# load model and scheduler
model_id = "stabilityai/stable-diffusion-x4-upscaler"
pipeline = StableDiffusionUpscalePipeline.from_pretrained(model_id, torch_dtype=torch.float16)
pipeline = pipeline.to("cuda")

# let's download an  image
url = "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/sd2-upscale/low_res_cat.png"
response = requests.get(url)
low_res_img = Image.open(BytesIO(response.content)).convert("RGB")
low_res_img = low_res_img.resize((128, 128))

prompt = "a white cat"

generator = torch.Generator(device="cuda").manual_seed(0)
upscaled_image = pipeline(prompt=prompt, image=low_res_img, generator=generator).images[0]
upscaled_image.save("upsampled_cat.png")


# test pt 


low_res_img_np = np.array(low_res_img).astype(np.float32) / 255.0
low_res_img_pt = torch.from_numpy(low_res_img_np.transpose(2,0,1))[None,:]
print(low_res_img_pt.min(), low_res_img_pt.max(), low_res_img_pt.shape)


generator = torch.Generator(device="cuda").manual_seed(0)
upscaled_image = pipeline(prompt=prompt, image=low_res_img_pt, generator=generator).images[0]
upscaled_image.save("upsampled_cat_pt.png")
```
![upsampled_cat](https://github.com/huggingface/diffusers/assets/12631849/107edf93-24be-42cf-a491-f37043005d12)

![upsampled_cat_pt](https://github.com/huggingface/diffusers/assets/12631849/65d791ea-47bf-46a1-9aca-5cf3d7631017)



If we pass a PT tensor of input range `[0,1]`, it will not work correctly, and not produce the same image of a PIL image input. That's because the current `preprocess` method will not normalize the image to `[-1,1]` if the it is PT tensor and we need to it be in `[-1,1]` for it to work 

```python
def preprocess(image):
    if isinstance(image, torch.Tensor):
        return image
    elif isinstance(image, PIL.Image.Image):
        image = [image]

    if isinstance(image[0], PIL.Image.Image):
        w, h = image[0].size
        w, h = (x - x % 64 for x in (w, h))  # resize to integer multiple of 64

        image = [np.array(i.resize((w, h)))[None, :] for i in image]
        image = np.concatenate(image, axis=0)
        image = np.array(image).astype(np.float32) / 255.0
        image = image.transpose(0, 3, 1, 2)
        image = 2.0 * image - 1.0
        image = torch.from_numpy(image)
    elif isinstance(image[0], torch.Tensor):
        image = torch.cat(image, dim=0)
    return image
```

**Comment by patrickvonplaten** [comment]
Cool think we can close this one, thanks for the iteration here @yiyixuxu
